src/distributions/slice.rs

```diff
@@ -9,9 +9,39 @@
 use core::num::NonZeroUsize;
 
 use crate::distributions::{Distribution, Uniform};
+use crate::Rng;
 #[cfg(feature = "alloc")]
 use alloc::string::String;
 
+#[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
+const _: () = assert!(false, "unsupported pointer width");
+
+#[derive(Debug, Clone, Copy)]
+enum UniformUsize {
+    U32(Uniform<u32>),
+    #[cfg(target_pointer_width = "64")]
+    U64(Uniform<u64>),
+}
```

Comment:
As with `gen_index`, this should probably be exposed somehow (left for a future PR).